onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -63,30 +65,39 @@ class NhwcTransformerImpl {
   std::deque<NodeIndex> removed_nodes_;
 };
 
-size_t NhwcTransformerImpl::RemoveOutputEdges(Node& node) {
-  size_t output_edges_count = node.GetOutputEdgesCount();
-  if (output_edges_count > 0) {
-    graph_utils::RemoveNodeOutputEdges(graph_, node);
-  }
-  // Bias the edge count to handle the case of a node that produces a graph
-  // output.
-  if (!graph_.GetNodeOutputsInGraphOutputs(node).empty()) {
-    output_edges_count++;
+// Caller make sure valid output_index
+size_t NhwcTransformerImpl::RemoveOutputEdge(Node& node, int output_index) {
+  size_t output_edges_count = graph_utils::RemoveNodeOutputEdges(graph_, node, output_index);
+
+  // Bias the edge count to handle the case of a node that produces a graph output.
+  auto node_outputs_for_graph = graph_.GetNodeOutputsInGraphOutputs(node);
+  for (auto idx : node_outputs_for_graph) {
+    if (output_index == idx) {
+      output_edges_count++;
+      break;
+    }
   }
   return output_edges_count;
 }
 
-void NhwcTransformerImpl::CreateNhwcArgument(Node& node, Node& nhwc_node, int rank) {
-  size_t original_uses = RemoveOutputEdges(node);
+void NhwcTransformerImpl::CreateNhwcArgument(Node& node, Node& nhwc_node, int rank, int output_index) {
+  size_t original_uses = RemoveOutputEdge(node, output_index);
 
   // Create a new NodeArg to track the output from the NHWC node.
   auto& output_defs = nhwc_node.MutableOutputDefs();
-  auto* output_original_arg = output_defs[0];
+  auto* output_original_arg = output_defs[output_index];
   std::string output_reorder_def_name = graph_.GenerateNodeArgName("reorder");
   auto* output_nhwc_arg = &graph_.GetOrCreateNodeArg(output_reorder_def_name, nullptr);
   nhwc_args_[output_original_arg] =
       onnxruntime::make_unique<NhwcArgument>(nhwc_node, output_nhwc_arg, original_uses, rank);
-  output_defs[0] = output_nhwc_arg;
+  output_defs[output_index] = output_nhwc_arg;
+}
+
+void NhwcTransformerImpl::CreateNhwcArgument(Node& node, Node& nhwc_node, int rank) {
+  int output_count = static_cast<int>(node.OutputDefs().size());
```

Comment:
change these ints to size_t instead? The graph_utils::GetNodeOutputEdges that you call above takes a size_t already, so that would be the more natural data type.

Reply:
It is just because the graph_util using int for output_index in most of their api(s). GetNodeOutputEdges() is not consistent with them. 
Anyway, Let me try to use size_t all in this file if possible.